### PR TITLE
[CI] Sync _log_softmax_batch_invariant with paddle update

### PR DESCRIPTION
**Pull request by EmmonsCurse**
## Motivation
Adapt to the API change of `paddle.nn.functional.log_softmax` introduced in https://github.com/PaddlePaddle/Paddle/pull/78220, where a new implementation is aligned and `paddle.compact.nn.functional.log_softmax` is introduced.  
The previous `_log_softmax_batch_invariant` implementation in FastDeploy did not fully align with the updated behavior, leading to potential incompatibility with the latest Paddle nightly version.

## Modifications

- Updated `_log_softmax_batch_invariant` to align with the latest `log_softmax` API behavior.
- Added support for the optional `out` parameter to maintain consistency with Paddle interfaces.
- Kept the original computation logic unchanged while ensuring compatibility with new Paddle updates.
- Ensured CI passes with the latest Paddle nightly builds.

## Usage or Command
N/A

## Accuracy Tests
N/A

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

**Review by ZhangYulongg — APPROVED** [review]

File: scripts/unittest_requirement.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ partial_json_parser
 jsonschema
 aistudio_sdk==0.3.5
 pandas
+use_triton_in_paddle
```
Comment (gongshaotian) [inline scripts/unittest_requirement.txt:11]: use_triton_in_paddle 这个依赖是不是不需要了

**Review by gongshaotian — APPROVED** [review]
LGTM

File: scripts/unittest_requirement.txt (hunk 1 above)
Comment (EmmonsCurse) [reply scripts/unittest_requirement.txt:11]: tests/model_executor/test_fused_moe_wint2_backend.py 这个单测需要这个依赖～
